tests/components/ActionButton/ActionButton.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { createSandbox, SinonSandbox, SinonSpy, spy } from 'sinon';
+import { Mock } from 'coveo-search-ui-tests';
+import { ActionButton, IActionButtonOptions } from '../../../src/components/ActionButton/ActionButton';
+import * as icons from '../../../src/utils/icons';
+
+describe('ActionButton', () => {
+    let sandbox: SinonSandbox;
+    let options: IActionButtonOptions;
+    let testSubject: ActionButton;
+
+    let appendIconSpy: SinonSpy;
+    let appendTitleSpy: SinonSpy;
+    let appendTooltipSpy: SinonSpy;
+
+    beforeAll(() => {
+        sandbox = createSandbox();
+    });
+
+    beforeEach(() => {
+        options = {
+            title: 'a default title',
+            tooltip: 'a default tooltip',
+            icon: icons.copy
+        };
+
+        testSubject = createActionButton(options);
+
+        appendIconSpy = sandbox.spy(<any>ActionButton.prototype, 'appendIcon');
```

Comment:
I don't agree with this way of testing.
I think you should assert the DOM.

My rule of thumb is that if we're breaking encapsulation to test, then there's a problem

Reply:
I agree that with complex components it's not necessarily a good idea to assert on the DOM, but when I see what you're trying to assert (i.e. Did I append the icon?) the DOM assertion make more sense.

(I was more frowning about the encapsulation break tbh)

Reply:
I get your point. You are right. There is no need to break encapsulation to validate the expectations here.